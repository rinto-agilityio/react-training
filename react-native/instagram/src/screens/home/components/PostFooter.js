import React from 'react'
import { View, Text } from 'react-native'
import PropTypes from 'prop-types'

// Components
import PostAction from './PostAction'
import CommentList from './CommentList'
import CommentInput from '@common/components/Comment'

// Styles
import CommonStyles from '@themes/common'
import { styles } from './styles/PostFooterStyles'

const PostFooter = ({
  isLiked,
  postId,
  likeCounting,
  comments,
  showComment,
  submitComment,
  owner,
  toogleLike,
  toogleCommentBox
}) => (
  <View style={styles.footer}>
    <PostAction
      isLiked={isLiked}
      postId={postId}
      userId={owner.id}
      toogleLike={toogleLike}
      toogleCommentBox={toogleCommentBox}
    />

    {likeCounting > 1 ? (
      <Text>{likeCounting} likes</Text>
    ) : (
      <Text>{likeCounting} like</Text>
    )}

    <CommentList comments={comments} />

    {showComment && (
      <CommentInput
        submitComment={submitComment}
        owner={owner}
        postId={postId}
      />
    )}
  </View>
)

PostFooter.defaultProps = {
  isLiked: false,
  postId: 0,
  likeCounting: 0,
  comments: [],
  showComment: false,
  owner: {}
}

PostFooter.propTypes = {
  isLiked: PropTypes.bool.isRequired,
  postId: PropTypes.number.isRequired,
  likeCounting: PropTypes.number.isRequired,
  comments: PropTypes.array.isRequired,
  showComment: PropTypes.bool.isRequired,
  owner: PropTypes.object.isRequired,
  submitComment: PropTypes.func,
  toogleLike: PropTypes.func,
  toogleCommentBox: PropTypes.func
}

export default PostFooter
