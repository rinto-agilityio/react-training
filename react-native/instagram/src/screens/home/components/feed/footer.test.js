// Components
import PostFooter from './Footer'

// Mocking data
import { photos } from '@test/__mocks__/sample-data'

describe('PostFooter component', () => {
  const emptyArrayLength = 0,
    singlePost = photos[1],
    defaultShowComment = false,
    mockSubmitComment = jest.fn(),
    mocktoggleLike = jest.fn(),
    mocktoggleCommentBox = jest.fn()

  it('Renders correctly without comment input', () => {
    const treeDOM = renderer
      .create(
        <PostFooter
          isLiked={singlePost.likes.length > emptyArrayLength}
          postId={singlePost.id}
          likeCounting={singlePost.likes.length}
          comments={singlePost.comments}
          owner={singlePost.owner}
          showComment={defaultShowComment}
          submitComment={mockSubmitComment}
          toggleLike={mocktoggleLike}
          toggleCommentBox={mocktoggleCommentBox}
        />
      )
      .toJSON()

    expect(treeDOM).toMatchSnapshot()
  })

  it('Renders correctly with comment input', () => {
    const treeDOM = renderer
      .create(
        <PostFooter
          isLiked={singlePost.likes.length > emptyArrayLength}
          postId={singlePost.id}
          likeCounting={singlePost.likes.length}
          comments={singlePost.comments}
          owner={singlePost.owner}
          showComment={!defaultShowComment}
          submitComment={mockSubmitComment}
          toggleLike={mocktoggleLike}
          toggleCommentBox={mocktoggleCommentBox}
        />
      )
      .toJSON()

    expect(treeDOM).toMatchSnapshot()
  })
})
