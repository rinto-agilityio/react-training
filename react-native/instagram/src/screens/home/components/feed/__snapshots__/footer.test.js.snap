// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostFooter component Renders correctly with comment input 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "padding": 10,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/heart-red.png",
          }
        }
        style={
          Object {
            "height": 24,
            "marginRight": 10,
            "width": 24,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    2 likes
  </Text>
  <View />
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
          "marginTop": 10,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://instagram.fdad3-2.fna.fbcdn.net/t51.2885-19/s150x150/19764837_461147117574705_5216900266812506112_a.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 11.25,
          "height": 22.5,
          "width": "5%",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      onSubmitEditing={[Function]}
      placeholder="Add a comment"
      style={
        Object {
          "borderColor": "#a7a7a7",
          "borderRadius": 15,
          "borderWidth": 1,
          "height": 30,
          "marginLeft": 5,
          "paddingLeft": 10,
          "width": "93%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`PostFooter component Renders correctly without comment input 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "padding": 10,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/heart-red.png",
          }
        }
        style={
          Object {
            "height": 24,
            "marginRight": 10,
            "width": 24,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    2 likes
  </Text>
  <View />
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
          "marginTop": 10,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://instagram.fdad3-2.fna.fbcdn.net/t51.2885-19/s150x150/19764837_461147117574705_5216900266812506112_a.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 11.25,
          "height": 22.5,
          "width": "5%",
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      onSubmitEditing={[Function]}
      placeholder="Add a comment"
      style={
        Object {
          "borderColor": "#a7a7a7",
          "borderRadius": 15,
          "borderWidth": 1,
          "height": 30,
          "marginLeft": 5,
          "paddingLeft": 10,
          "width": "93%",
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
